File: Basic Codes/TrafficSimulator2/src/VehicleLoad.java
import java.util.*;
import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;
import java.util.Vector;
import javax.swing.*;

public class VehicleLoad {
    
    String fileName1 = "Path1.txt";
    String fileName2 = "Demand1.txt";

    private List<Path> pathList= new ArrayList<Path>();
    private List<Path> pathRemovalList= new ArrayList<Path>();
    private List<Demand> demandList= new ArrayList<Demand>();

    public VehicleLoad()
    {
        readPath();
        readDemand();
        addPath();
    }
    public void readPath()
    {
        String DataLine = "";
        try
        {
            File inFile = new File(fileName1);
            BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(inFile)));

            int index;

            DataLine = br.readLine();
            int numPaths = Integer.parseInt(DataLine);

            for(index =0; index<numPaths; index++)
            {
            	DataLine = br.readLine();
            	StringTokenizer stToken = new StringTokenizer(DataLine," ");
            	int nodeId1 = Integer.parseInt(stToken.nextToken());
            	int nodeId2 = Integer.parseInt(stToken.nextToken());
                pathList.add(new Path(nodeId1,nodeId2));
                while(stToken.hasMoreTokens())
                {
                    int linkIndex = Integer.parseInt(stToken.nextToken());
                    pathList.get(index).addLinkIndex(linkIndex);
                }
            }         
            br.close();
        }
        catch (FileNotFoundException ex)
        {
            System.out.println("FileNotFoundException HD");
        }
        catch (IOException ex)
        {
        	System.out.println("exception");
        }
    }
    public void readDemand()
    {
        String DataLine = "";
        try
        {
            File inFile = new File(fileName2);
            BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(inFile)));

            int index;

            DataLine = br.readLine();
            int numDemands = Integer.parseInt(DataLine);

            for(index =0; index<numDemands; index++)
            {
            	DataLine = br.readLine();
            	StringTokenizer stToken = new StringTokenizer(DataLine," ");
            	int nodeId1 = Integer.parseInt(stToken.nextToken());
            	int nodeId2 = Integer.parseInt(stToken.nextToken());
                int demand = Integer.parseInt(stToken.nextToken());
                //demand=1;
                demandList.add(new Demand(nodeId1,nodeId2,demand));
            }         
            br.close();
        }
        catch (FileNotFoundException ex)
        {
            System.out.println("FileNotFoundException HD");
        }
        catch (IOException ex)
        {
        	System.out.println("exception");
        }
    }
    public void addPath()
    {
        int i,j,k;
        for(i=0;i<demandList.size();i++)
        {
            for(j=0;j<pathList.size();j++)
            {
                if(pathList.get(j).correctPath(demandList.get(i).getSource(),demandList.get(i).getDestination()))
                {
                    demandList.get(i).addPath(pathList.get(j));
                    pathRemovalList.add(pathList.get(j));
                }
            }
            for(k=0;k<pathRemovalList.size();k++)
            {
                pathList.remove(pathRemovalList.get(k));
            }

        }
    }
    public List<Demand> getDemand()
    {
        return demandList;
    }
}

File: Basic Codes/DhakaSim-qt/qt/qt/userinterface_prev/vehicleload.cpp
#include "vehicleload.h"
#include"others.h"
#include<stdlib.h>
#include<stdio.h>
#include<QApplication>
VehicleLoad::VehicleLoad()
{
    readPath();
    readDemand();
    addPath();
}

void VehicleLoad::readPath()
{
    try{
    FILE *fp = fopen(filename1.c_str(), "r");
    int index,numPaths,nodeId1,nodeId2;

    fscanf(fp,"%d",&numPaths);
    char str[50];
    for(index = 0; index<numPaths; index++)
    {
        fscanf(fp,"%c",&str[0]);  //consumes the \n character
        fscanf(fp,"%[^\n]",str);
        char *pp;
        pp = strtok(str," ");
        vector<string> V;
        while(pp!=NULL)
        {
            V.push_back(pp);
            pp = strtok(NULL," ");
        }

        nodeId1 = atoi(V[0].c_str());
        nodeId2 = atoi(V[1].c_str());
        Path pt(nodeId1,nodeId2);
        pathList.push_back(pt);

        for(int x=2; x < V.size(); x++)
        {
            int linkId = atoi(V[x].c_str());
            pathList[index].addLinkId(linkId);
        }

    }
    fclose (fp);

    }catch(string s){}

}

void VehicleLoad::readDemand()
{
    FILE *fp = fopen(filename2.c_str(), "r");
    int index,numDemands,nodeId1,nodeId2,demand;
    fscanf(fp,"%d",&numDemands);
    char ch;
    for(index =0; index < numDemands; index++)
    {
        fscanf(fp,"%c",&ch);   //consumes the \n character
        fscanf(fp,"%d%d%d",&nodeId1,&nodeId2,&demand);
        Demand dm(nodeId1,nodeId2,demand);
        demandList.push_back(dm);
    }

    fclose(fp);

}

void VehicleLoad::addPath()
{
    int i,j;
    for(i=0; i<demandList.size(); i++)
    {
        for(j=0;j<pathList.size(); j++)
        {
            qDebug("j:%d",j);
            if(pathList[j].correctPath(demandList[i].getSource(),demandList[i].getDestination()))
            {

                demandList[i].addPath(pathList[j]);
                pathList.erase(pathList.begin()+j);
                j--;
            }
        }
    }
}
vector <Demand> VehicleLoad::getDemand()
{
    return demandList;
}





